facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -241,39 +269,103 @@ private org.jboss.pnc.model.Artifact getPncArtifact(Artifact art) {
+        Path path = Paths.get(artifact.getFilename());
+        builder.filename(path.getFileName().toString());
+        builder.identifier(artifact.getFilename());
+        Path directory = path.getParent();
+        builder.deployPath(directory == null ? null : directory.toString());
+
+        return builder.build();
+    }
+
+    private org.jboss.pnc.model.Artifact mapBrewArtifact(Artifact artifact, String nvr) {
+        if (artifact.getArtifactType() != ArtifactType.MAVEN) {
+            throw new UnsupportedOperationException("Brew artifact " + artifact + " is not Maven!");
         }
-        if (art.isBuiltFromSource()) {
+        MavenArtifact mavenArtifact = (MavenArtifact) artifact;
+
+        org.jboss.pnc.model.Artifact.Builder builder = mapArtifact(mavenArtifact);
+        builder.identifier(createIdentifier(mavenArtifact));
+        builder.filename(createFileName(mavenArtifact));
+        builder.deployPath(createDeployPath(mavenArtifact));
+        builder.originUrl(createBrewOriginURL(mavenArtifact, nvr));
+        builder.purl(createPURL(mavenArtifact));
+
+        return builder.build();
+    }
+
+    private org.jboss.pnc.model.Artifact.Builder mapArtifact(Artifact artifact) {
+        org.jboss.pnc.model.Artifact.Builder builder = org.jboss.pnc.model.Artifact.builder();
+        builder.md5(artifact.getMd5());
+        builder.sha1(artifact.getSha1());
+        builder.sha256(artifact.getSha256());
+        builder.size(artifact.getSize());
+        builder.importDate(new Date());
+        builder.creationUser(userService.currentUser());
+        builder.creationTime(new Date());
+
+        if (artifact.isBuiltFromSource()) {
             builder.artifactQuality(ArtifactQuality.NEW);
         } else {
             builder.artifactQuality(ArtifactQuality.IMPORTED);
         }
 
-        return builder.build();
+        return builder;
     }
 
-    private String fill(MavenArtifact mavenArtifact) {
+    private static String createDeployPath(MavenArtifact mavenArt) {
+        String filename = createFileName(mavenArt);
+        String deployPath = mavenArt.getGroupId().replace('.', '/') + "/" + mavenArt.getArtifactId() + "/"
+                + mavenArt.getVersion() + "/" + filename;
+        return deployPath;
+    }
+
+    private static String createFileName(MavenArtifact mavenArt) {
+        String filename = mavenArt.getArtifactId() + "-" + mavenArt.getVersion();
+        if (Strings.isEmpty(mavenArt.getClassifier())) {
+            filename += "-" + mavenArt.getClassifier();
+        }
+        filename += "." + mavenArt.getType();
+        return filename;
+    }
+
+    private String createBrewOriginURL(MavenArtifact mavenArt, String nvr) {
+        String brewContentUrl = globalConfig.getBrewContentUrl();
+
+        Matcher matcher = NVR_PATTERN.matcher(nvr);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException("NVR " + nvr + " does not match expected format.");
```

Comment:
Wouldn't it be better to log an error and return null instead? Missing the originUrl is not a big deal - we don't really use it - and can be corrected later.

Reply:
this shuldn't really happen at all, unless there is a big bug somewhere